lib/zendesk_apps_support/validations/mime.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module ZendeskAppsSupport
+  module Validations
+    module Mime
+      UNSUPPORTED_MIME_TYPES = %w[
+        vnd.rar rar zip gzip pdf doc docx avi bin bz bz2 csh sh jar mp3 mpeg odt pptx ppt xls xlsx 7z
+      ].freeze
+
+      class << self
+        def call(package)
+          unsupported_files = package.files.find_all do |app_file|
+            identify_mime_type(app_file) && block_listed?(app_file)
```

Comment:
Why do we need `identify_mime_type(app_file) &&` here?

Reply:
Because when we were using MimeMagic, we identify that gems might not be able to identify all mime types. 

I was thinking, if they are not identify-able as one of the UNSUPPORTED_MIME_TYPES, it shouldn't matter. Also works as a precaution. 

Curious to know what do you think?

Reply:
As discussed in meat space,

we have decided to do an
`early return if mime type is mime_type.nil?` or `early return unless mime_type`

Reply:
Update:

Reverting back to MimeMagic because it opens files to evaluate content.
```
mime_type = MimeMagic.by_magic(app_file.read)

content_subtype = mime_type.subtype if mime_type
extension_name = app_file.extension.delete('.')

([content_subtype, extension_name] & UNSUPPORTED_MIME_TYPES).any?
```
With the lines above, we do not need to worry if content subtype can't be identified, content_subtype can be nil. Also, we check for `app_file.extension` as well.